Framework/Crystal/src/SaveHKL.cpp

```diff
@@ -441,11 +441,23 @@ void SaveHKL::exec() {
         if (cosines) {
           out << std::setw(8) << std::fixed << std::setprecision(5) << lambda;
           out << std::setw(8) << std::fixed << std::setprecision(5) << tbar;
-          Kernel::DblMatrix oriented = p.getGoniometerMatrix();
+          Kernel::DblMatrix R = p.getGoniometerMatrix();
 
-          auto U = ol.getU();
-          auto RU = oriented * U;
-          RU.Transpose();
+          R.Transpose();
+
+          DblMatrix T = ol.getUB();
+
+          V3D t1 = T * V3D(1, 0, 0);
+          V3D t2 = T * V3D(0, 1, 0);
+          V3D t3 = T * V3D(0, 0, 1);
+
+          t1.normalize();
+          t2.normalize();
+          t3.normalize();
+
+          T.setRow(0, t1);
+          T.setRow(1, t2);
+          T.setRow(2, t3);
```

Comment:
Is this equivalent to transposing `T` (after normalising columns)? 
If one made a copy of the oriented lattice, then set `a = b = c = 1` would the transpose of `UB` then be the same as this matrix `T`?

Reply:
This is the transpose. Your second statement would be correct as it would be the identity.